ceno_zkvm/src/tables/ram.rs

```diff
@@ -13,15 +13,21 @@ pub struct MemTable;
 impl DynVolatileRamTable for MemTable {
     const RAM_TYPE: RAMType = RAMType::Memory;
     const V_LIMBS: usize = 1; // See `MemoryExpr`.
-    const OFFSET_ADDR: Addr = CENO_PLATFORM.ram_start();
-    const END_ADDR: Addr = CENO_PLATFORM.ram_end() + 1;
+
+    fn offset_addr(platform: &Platform) -> Addr {
+        platform.ram_start()
+    }
+
+    fn end_addr(platform: &Platform) -> Addr {
+        platform.ram_end()
+    }
 
     fn name() -> &'static str {
         "MemTable"
     }
 
-    fn max_len() -> usize {
-        let max_size = (Self::END_ADDR - Self::OFFSET_ADDR) / WORD_SIZE as Addr;
+    fn max_len(platform: &Platform) -> usize {
+        let max_size = (Self::end_addr(platform) - Self::offset_addr(platform)) / WORD_SIZE as Addr;
```

Comment:
This is inclusive end (possibly `u32::MAX`). More reliable is:

```suggestion
        let max_size = (Self::end_addr(platform) - Self::offset_addr(platform)).div_ceil(WORD_SIZE as Addr);
```